pkg/p2p/p2p.go

```diff
@@ -14,15 +14,20 @@ import (
 
 type Service interface {
 	AddProtocol(ProtocolSpec) error
-	Connect(ctx context.Context, addr ma.Multiaddr) (overlay swarm.Address, err error)
-	Disconnect(overlay swarm.Address) error
-	Peers() []Peer
+	Connecter
 }
 
 type Streamer interface {
 	NewStream(ctx context.Context, address swarm.Address, protocol, version, stream string) (Stream, error)
 }
 
+type Connecter interface {
```

Comment:
This doesn't really sit well with go interface naming conventions. `Connecter` would be a name for an interface that defines only a `Connect` method. Service might be better in this case